### ISSUE TITLE
Oneserver option with --server specified still connects to remote servers

### ISSUE THREAD

**Issue by atweiden** [spesmilo/electrum#992]
I noticed this issue when testing an electrum-server instance on localhost.

```sh
$ electrum -s localhost:50001:t -1
```

It used to only connect to the localhost server, but continues to connect to remote hosts now. The `-1` option does not seem to have an effect.

**Comment by ecdsa** [comment]
the -1 option works for me, but I did not try on localhost.
does the problem occur only with localhost ?

**Comment by atweiden** [comment]
The more concerning part is the `-s` cmdline option failing to be read from. To specify localhost server and then to connect to remote servers should not happen.

This occurs with remote servers specified by `-s` too. I remember this happening to me back when I was testing [electrum-2of3](https://github.com/atweiden/electrum-2of3).

Yesterday, I solved it for localhost by opening Network settings in QT, unchecking autoconnect and manually writing in localhost:50001. Maybe it is something to do with ~/.electrum/config.

The `-1` option may actually work. I didn't check for "downloading blockheaders from 1 servers". I thought `-1` was supposed to limit all connections to the server specified by `-s`, that's all. When specifying `-s -1` for localhost the client should not be making remote network connections.

**Comment by ecdsa** [comment]
can you reproduce this reliably?
I suspect a more general problem, maybe with your config file..

**Comment by atweiden** [comment]
Yes. It happens again whenever GUI network preferences are set to autoconnect to a server, but `-s` is passed on cmdline. Getting this now:

```sh
$ electrum -s localhost:50001:t
Warning: not changing key 'server' because it is not modifiable (passed as command line option or defined in /etc/electrum.conf)
```

(it connected to electrum.be)

Oddly the `-1` option seems to enforce the `-s` option, but it failed to do that earlier.

[~/.electrum/config](http://ix.io/g2x)

**Comment by ecdsa** [comment]
What you describe in your last post differs from your first post, and is expected behavior. 
`electrum -s myserver` will connect to `myserver` in order to request your balance, and to 8 other nodes to request block headers.
`electrum -s myserver -1` will connect only to `myserver`

**Comment by atweiden** [comment]
It does not request balance from localhost. My localhost electrum-server was offline when I tested this today. It skips connecting to localhost and connects to remote servers with `electrum -s localhost:50001:t`.

For `electrum -s localhost:50001:t -1`, oddly the `-1` flag makes it not connect to remote servers to update balances at all. It stays offline and unconnected.

However, even `electrum -s localhost:50001:t -1` didn't work for me on the day I opened this issue.
